compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -603,7 +604,11 @@ bool ResolutionCandidate::checkResolveFormalsWhereClauses(CallInfo& info) {
 
       bool promotes          = false;
 
-      if (actualIsTypeAlias != formalIsTypeAlias) {
+      if (isInitCopy && isString(actual) && formal->getValType() == dtStringC) {
+        // Do not allow an initCopy of a string to find the c_string initCopy,
+        // which is considered first because it is not compiler generated.
```

Comment:
Would it be harmful to mark the c_string initCopy as "last resort"?  Or would it just not resolve the problem?

Reply:
I initially tried that, but when it didn't work I just did this. I'm not very familiar with the last-resort stuff, so I may have messed something up.

Reply:
It's probably because the compiler generated initializer for strings is also marked last resort so it thinks there is an ambiguity due to the coercion.  Oh well